src/index.ts

```diff
@@ -50,17 +52,145 @@ export default class WorkspaceClient {
     }
 
     private static createAxiosInstance(config: IRestAPIConfig): AxiosInstance {
-        if (config.ssCrtPath && this.isItNode() && fs.existsSync(config.ssCrtPath)) {
-            const agent = new https.Agent({
-                ca: fs.readFileSync(config.ssCrtPath)
-            });
-            return axios.create({httpsAgent: agent});
+        if (! this.isItNode()) {
+            return axios;
+        }
+
+        let certificateAuthority : Buffer | undefined;
+        if (config.ssCrtPath && fs.existsSync(config.ssCrtPath)) {
+            certificateAuthority = fs.readFileSync(config.ssCrtPath);
         }
 
+        const proxyUrl = process.env.http_proxy;
+        if (proxyUrl && proxyUrl !== '' && config.baseUrl) {
+            let mainProxyOptions: tunnel.ProxyOptions;
+            var parsedProxyUrl = url.parse(proxyUrl);
+            let port = 3128;
+            if (parsedProxyUrl.port && parsedProxyUrl.port !== '') {
+                port = Number(parsedProxyUrl.port);
```

Comment:
The result of parsing a string as a number could be `NaN`. So you need additionally check if `port` is not `NaN`.